### PR TITLE
[SPARK-41095][SQL] Convert unresolved operators to internal errors

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to interpret the `unresolved operator` issue as an internal error, and throw `SparkException` w/ the error class `INTERNAL_ERROR`.

### Why are the changes needed?
The issues that leads to `unresolved operator` should be solved earlier w/ proper user-facing errors. If we reach the point when we cannot resolve an operator, we should interpret this as Spark SQL internal error.

### Does this PR introduce _any_ user-facing change?
No, in most regular cases.

### How was this patch tested?
By running the affected and modified test suites:
```
$ PYSPARK_PYTHON=python3 build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
$ build/sbt "test:testOnly *AnalysisErrorSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @srielau @itholic @panbingkun @LuciferYang Could you review this PR, please.

File: core/src/main/scala/org/apache/spark/SparkException.scala [hunk 1]
```diff
@@ -68,6 +68,17 @@ class SparkException(
 }
 
 object SparkException {
+  def internalError(msg: String, context: Array[QueryContext]): SparkException = {
+    val errorClass = "INTERNAL_ERROR"
+    val messageParameters = Map("message" -> msg)
+    new SparkException(
+      message = SparkThrowableHelper.getMessage(errorClass, messageParameters),
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/SparkException.scala:75]: shall we change other `internalError` overloads to load template from json file?

**Review by LuciferYang — APPROVED** [review]
fine to me

File: core/src/main/scala/org/apache/spark/SparkException.scala (hunk 1 above)
Comment (MaxGekk) [reply core/src/main/scala/org/apache/spark/SparkException.scala:75]: @cloud-fan Could you clarify this `to load template from json file`, please. All `internalError` load templates from `error-classes.json`, as I can see `internalError` invoke `SparkThrowableHelper.getMessage()` eventually.

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan and @LuciferYang for review.

**Pull request closed without merging**